optimum/habana/sentence_transformers/st_gaudi_trainer.py

```diff
@@ -262,11 +262,17 @@ def call_model_init(self, trial=None) -> "SentenceTransformer":
     def override_model_in_loss(self, loss: torch.nn.Module, model: "SentenceTransformer") -> torch.nn.Module:
         from sentence_transformers import SentenceTransformer
 
-        for name, child in loss.named_children():
-            if name == "model" and isinstance(child, SentenceTransformer):
+        # Directly replace the model if it exists and is not already the wrapped model
+        if hasattr(loss, "model") and isinstance(loss.model, SentenceTransformer):
+            # Check if the current model is already the wrapped model to prevet redundant assignments
+            if loss.model != model:
                 loss.model = model
-            elif isinstance(child, torch.nn.Module):
-                setattr(loss, name, self.override_model_in_loss(child, model))
+        # Recursively traverse child modules without replacing already wrapped models
+        for name, child in loss.named_children():
+            if isinstance(child, torch.nn.Module):
+                # Avoid replacing the model again if it's already the desired model
+                if not (name == "model" and child is model):
```

Comment:
```suggestion
                if name != "model" or child is not model:
```

can we simplify the condition! (Personal suggestion)